vlasovsolver/vlasovmover.cpp

```diff
@@ -348,7 +348,14 @@ void calculateSpatialTranslation(
    
    // Mapping complete, update moments and maximum dt limits //
 momentCalculation:
-   calculateMoments_R(mpiGrid,localCells,true);
+   if(P::isRestart && P::tstep_min == P::tstep && dt == 0.0)
```

Comment:
So I think that instead of this check, we should revert the changes at 
https://github.com/markusbattarbee/vlasiator-public/blob/25efbf93afae067b246aef3550f4bb46c1693860/vlasiator.cpp#L518
So that zero-timestep translation and acceleration are performed only if it's not a restart.

Reply:
Right, no notification from github for me, still :(. That change had the purpose of preserving the population-specific moment calculations, otherwise the first bulk output after restart is garbled for pop moments. Not very relevant, since the moment recalculation error is small anyhow.

Reply:
Ok. That's a bit of a special case where output is triggered right after a restart happened and before a single propagation has taken place - unlikely, but can indeed happen. (And I think what'd happen is that file would have already been written before the restart, but would get overwritten by the new run)

This however is something that could be tackled by re-ordering or adding a momentscalculation call to output if we're at the first timestep instead of adding lots of ifs inside the momentcalculation function. And should be fixed in regular Vlasiator as well! :)

Reply:
Yes, that seems a more reasonable solution.